test/Database/LSMTree/Model/Session.hs

```diff
@@ -844,7 +844,9 @@ supplyUnionCredits ::
   -> UnionCredits
   -> m UnionCredits
 supplyUnionCredits t@Table{..} c@(UnionCredits credits)
-  | credits <= 0 = pure c
+  | credits <= 0 = do
+      _ <- guardTableIsOpen t
+      pure c
```

Comment:
It's a bit more consistent/regular to specify it this way. We don't _really_ expect the impl to allow supplying 0 credits to closed tables without an error. All operations on closed tables are an error, trivial or not.